bin/database.py

```diff
@@ -1054,6 +1054,75 @@ def upgrade_to_29():
     users = config.db.users.find({})
     process_cursor(users, upgrade_to_29_closure)
 
+def upgrade_to_30_closure_analysis(session):
+    analyses = session.get('analyses', None)
+
+    if analyses is not None:
+        for analysis_ in analyses:
+            files = analysis_.get('files', [])
+            for file_ in files:
+                if 'created' not in file_:
+                    file_['created'] = analysis_['created']
```

Comment:
Given unexpected missing fields on other objects, I wouldn't assume `analysis_['created']` exists. I don't have a strong opinion on what the fallback should be if that's the case. Using something like epoc would allow easy identification in the future.

Same comment for the other closure methods. Test both present and missing `created` source keys, and ensure the db update completes fully.